db/osqlcomm.c

```diff
@@ -7096,84 +7058,49 @@ void signal_replicant_error(const char *host, unsigned long long rqid,
                          int nettype)
 {
-    struct ireq *iq = NULL;
-    int rc = 0;
-    uint8_t *malcd = malloc(OSQL_BP_MAXLEN);
-    if (!malcd)
-        goto done;
-
     osql_sess_t *sess = NULL;
-    osql_req_t req;
-    bool is_reorder_on = false;
-    uint8_t *p_req_buf = dtap;
-    const uint8_t *p_req_buf_end = p_req_buf + dtalen;
-    uint8_t *p_buf = malcd;
-    const uint8_t *p_buf_end = p_buf + OSQL_BP_MAXLEN;
-    char *sql;
-    char *sqlret = NULL;
-    int sqllenret = 0;
-    int debug = 0;
+    unsigned long long rqid;
     uuid_t uuid;
-    const char *errmsg = "";
+    char *sql;
+    int sqllen;
+    char *tzname;
+    int flags = 0;
     int send_rc = 1;
+    const char *errmsg = "";
+    int rc = 0;
 
     /* grab the request */
+    uint8_t *p_req_buf = dtap;
+    const uint8_t *p_req_buf_end = p_req_buf + dtalen;
     if (osql_nettype_is_uuid(nettype)) {
-        osql_uuid_req_t uuid_req;
-        sql = (char *)osqlcomm_req_uuid_type_get(&uuid_req, p_req_buf,
-                                                 p_req_buf_end);
-        req.type = uuid_req.type;
-        req.rqlen = uuid_req.rqlen;
-        req.sqlqlen = uuid_req.sqlqlen;
-        req.rqid = OSQL_RQID_USE_UUID;
-        req.flags = uuid_req.flags; // NB: we will loose the 0x80 and up flags
-        memcpy(req.tzname, uuid_req.tzname, sizeof(uuid_req.tzname));
-        comdb2uuidcpy(uuid, uuid_req.uuid);
-        is_reorder_on = ((uuid_req.flags & OSQL_FLAGS_REORDER_ON) != 0);
+        osql_uuid_req_t ureq;
+        sql =
+            (char *)osqlcomm_req_uuid_type_get(&ureq, p_req_buf, p_req_buf_end);
+        rqid = OSQL_RQID_USE_UUID;
+        comdb2uuidcpy(uuid, ureq.uuid);
+        flags = ureq.flags;
+        tzname = ureq.tzname;
+        sqllen = ureq.sqlqlen;
     } else {
+        osql_req_t req;
         sql = (char *)osqlcomm_req_type_get(&req, p_req_buf, p_req_buf_end);
+        rqid = req.rqid;
         comdb2uuid_clear(uuid);
-    }
-
-    if (!p_buf) {
-        logmsg(LOGMSG_ERROR, "%s:unable to allocate %d bytes\n", __func__,
-                OSQL_BP_MAXLEN);
-        errmsg = "unable to request buffer";
-        rc = -1;
-        goto done;
-    }
-
-    /* construct a block transaction */
-    if (osql_bplog_build_sorese_req(p_buf, &p_buf_end, sql, req.sqlqlen,
-                                    req.tzname, type, &sqlret, &sqllenret,
-                                    req.rqid, uuid)) {
-        logmsg(LOGMSG_ERROR, "bug in code %s:%d", __func__, __LINE__);
-        errmsg = "bug in code";
-        rc = -1;
-        goto done;
+        flags = req.flags;
+        tzname = req.tzname;
+        sqllen = req.sqlqlen;
     }
 
     /* create the request */
-    sess = osql_sess_create(sqlret, sqllenret, req.tzname, type, req.rqid, uuid,
-                            fromhost, malcd, is_reorder_on);
+    sess = osql_sess_create(sql, sqllen, tzname, type, rqid, uuid, fromhost,
+                            flags & OSQL_FLAGS_REORDER_ON);
```

Comment:
flags & OSQL_FLAGS_REORDER_ON should be set only in the case of uuid net type -- in case that flag is accidentally on for rqid type, we should simply ignore it. I think this caused issues in my testing for rqid messages -- if you see any issues in robomark, we might want to leave it the way the parameter was passed before.

Reply:
it is set on replicant; bplog has no knowledge of the protocol replicant use when creating the bplog.  ordering the bplog indexes using different criteria's is still on ordering, and as long as it is global, it is serving the same purpose of serializing the writers with the purpose of avoiding deadlocks.